File: Data Structure/Stack/acmicpc.net/1406/Main.py
import sys

# input = sys.stdin.readline
file = open("./01 ing/acmicpc.net/sample2.txt", "r")
input = file.readline

left = list(input().strip())
right = []
n = int(input())
for _ in range(n):
    operation = input().split();
    if(operation[0] == "P"):
        left.append(operation[1]);
    elif(operation[0] == "L" and len(left) !=0):
        right.append(left.pop());
    elif(operation[0] == "D" and len(right) !=0):
        left.append(right.pop());
    elif(operation[0] == "B" and len(left) !=0):
        left.pop();

ans=[]
for i in range(len(left)):
    ans.append(left[i]);
for i in range(len(right)):
    ans.append(right.pop());

print("".join(ans))

File: Data Structure/Stack/acmicpc.net/1406/Main.js
"use strict";

const inputFilePath = "/sample.txt";
const input = require("fs")
  .readFileSync(__dirname + inputFilePath)
  .toString()
  .trim()
  .split("\n");
// const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");

const left = input[0].split("");
const right = [];
const n = input[1];
for (let i = 0; i < n; i++) {
  const cmd = input[i + 2].split(" ");
  if (cmd[0] === "P") {
    left.push(cmd[1]);
  } else if (cmd[0] === "L" && left.length !== 0) {
    right.push(left.pop());
  } else if (cmd[0] === "D" && right.length !== 0) {
    left.push(right.pop());
  } else if (cmd[0] === "B" && left.length !== 0) {
    left.pop();
  }
}
const ans = left.concat(right.reverse());

console.log(ans.join(""));
